app/models/custom_field.rb

```diff
@@ -14,4 +14,20 @@ class CustomField < ActiveRecord::Base
              foreign_key: 'last_modified_by_id',
              class_name: 'User'
   has_many :sample_custom_fields, inverse_of: :custom_field
+
+  after_create :update_samples_table_state
+
+  def update_samples_table_state
```

Comment:
It would be more DRY if you would combine this within `create_samples_table_state`, and just call this method instead and do some logic branching etc.